AsyncDisplayKit/ASMapNode.mm

```diff
@@ -159,6 +161,15 @@ - (void)setRegion:(MKCoordinateRegion)region
 
 - (void)takeSnapshot
 {
+  // do not trigger snapshot before layout
+  ASLayout *layout = self.calculatedLayout;
+  if (layout == nil || CGSizeEqualToSize(CGSizeZero, layout.size)) {
```

Comment:
@aaronschubert0    Let me know if Hannah's description doesn't address your concern. This approach seems to make sense to me, or at least I am not sure what a better route would be to addressing the very real crash reported here.

@garrettmoon   Do you think it would be more clear with a different instance variable name? Perhaps mentioning something about a deferred snapshot?

I agree, the comment could definitely be improved to describe that the condition is critical to avoid crashes, although it doesn't seem that opaque to me -- from the perspective that, at least as a performance optimization, a zero sized snapshot won't do any good.  The level of criticality is less clear though.